generators/reference_chunking/newline_splitter/__init__.py

```diff
@@ -0,0 +1,22 @@
+from pydantic import BaseModel
+
+INPUT_EXAMPLE = {
+    "text": """This is the first line.
+    And this is the second line.
+    Here's a third one, too.
+    """,
+}
+
+class NewlineSplitterModel(BaseModel):
+    text: str
+
+    class Config:
+        schema_extra = {"example": INPUT_EXAMPLE}
+
+
+def newline_splitter(req: NewlineSplitterModel):
+    """Splits a text by newline characters"""
+    splits = req.text.split("\n")
+    return {"splitted_text" : [val.strip() for val in splits if len(val) > 0]}
```

Comment:
this will still add a line with spaces as a chunk since we check the length before stripping.
To prevent using strip twice in the list comprehension i think we should strip on creation

```suggestion
    splits = [t.strip() for t in req.text.split("\n")]
    return {"splitted_text" : [val for val in splits if len(val) > 0]}
```

Reply:
Makes sense, added it.